src/dpl/src/Grid.cpp

```diff
@@ -156,6 +239,16 @@ void Opendp::visitCellPixels(
   dbMaster* master = inst->getMaster();
   auto obstructions = master->getObstructions();
   bool have_obstructions = false;
+
+  int site_width, row_height;
+  if (isStdCell(&cell)) {
```

Comment:
usually, they are very big "blocks" or "cells" that don't have their own separate 2D grid in the 3D grid, so it is enough to keep them to the single-height grid and we mark all layers anyway. 
I am not sure if we should legalize them or not, but since I didn't change the core logic, I assume they were always legalized.